### PR TITLE
Creation wizard lib tests

### PR DESCRIPTION
**Pull request by m4dz**
Tests for the Account creation wizard lib:

- [x] validateState
- [x] sanitizeConfig
- [x] parseProviders

### CODE REVIEW

File: client/test/accounts_lib.spec.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+const AccountsLib = require('../app/libs/accounts')
+
+
+describe("Accounts libs spec", () => {
+
+    describe("Validate and returns a new state", () => {
+
+        it("should returns a simple value in a new state", () => {
+            let state = {}
+            let nextState = AccountsLib.validateState('login', 'mail@cozy.io', state)
```
Comment (jsilvestre) [inline client/test/accounts_lib.spec.js]: Out of curiosity: why do you use let instead of const here?

Comment (m4dz) [inline client/test/accounts_lib.spec.js]: Coz' in some tests I need to change their values (e.g. https://github.com/cozy/cozy-emails/pull/866/files/769ab196b3f04156aa8bbc009ccf4ce34e6f3779#diff-75f7b93415035b03c811d0173ae2f7fbR58) so I try to stay consistent and use `let` everywhere. But I'm open to another solution if you provide one :smile:

Comment (jsilvestre) [inline client/test/accounts_lib.spec.js]: That's what I thought, I was not sure I would have kept with consistency in this case. I wondered if there was something else. Thanks!

File: client/test/accounts_lib.spec.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            assert.propertyVal(nextState, 'imapLogin', nextState.login)
+            assert.propertyVal(nextState, 'smtpLogin', nextState.login)
+
+            nextState = AccountsLib.validateState('login', 'box@cozy.io', nextState)
+
+            assert.propertyVal(nextState, 'imapLogin', nextState.login)
+            assert.propertyVal(nextState, 'smtpLogin', nextState.login)
+        })
+
+        it("should leave imap / smtp login properties untouched when they differ from login", () => {
```
Comment (misstick) [inline client/test/accounts_lib.spec.js]: @m4dz : shouldnt you use a new group such as: `imap / smtp properties` for more lisibility?

**Comment by misstick** [comment]
@m4dz : I don't see the point of prefixing commit with `[test]` because its the goal of this PR; I think that you can a flag to the PR, such as `test`. This is done form `enhancement` or `bug`, why not for tests?

Look at my previous comment to add more lisibility to your test; otherwhise everything looks good to me! 👍

**Comment by m4dz** [comment]
>  I don't see the point of prefixing commit with [test]

Because of https://github.com/commitizen/cz-cli#using-the-command-line-tool :smiley:. We don't use explicitely Commitizen, but I use their prefix convention for commits.

**Pull request merged**